console-subscriber/src/builder.rs

```diff
@@ -376,6 +389,12 @@ impl Builder {
         thread::Builder::new()
             .name("console_subscriber".into())
             .spawn(move || {
+                let _subscriber_guard;
+                if !self_trace {
+                    _subscriber_guard = tracing::subscriber::set_default(
+                        tracing_core::subscriber::NoSubscriber::default(),
+                    );
+                }
```

Comment:
nit, take it or leave it: this approach seems fine to me. a slightly more concise alternative could be to use [`bool::then`](https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then) to create an `Option<DefaultGuard>`:
```suggestion
                let _subscriber_guard = self_trace.then(|| {
                    tracing::subscriber::set_default(tracing_core::subscriber::NoSubscriber::default())
                });
```
but the current implementation might be clearer, so feel free to take it or leave it!

Reply:
Interesting! Maybe not clear enough as the test is negated.